*: remove all version checks against 23.2

RaduBerinde opened the pull request:
Now that `MinSupported` is `V23_2`, we no longer need to check if 23.2
is active. This change removes all these checks.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/124372)
<!-- Reviewable:end -->

RaduBerinde commented:
Friendly ping

celiala reviewed (approved):
Appreciate the clean up! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 36 of 36 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124372#-:-NyU0TVf0JZsXWFGIjIA:b-1ew1ov)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dt, @michae2, @nameisbhaskar, @rail, and @vidit-bhat)



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
lgtm!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124372#-:-NyXUHnxER2EV5klr5eR:b-7nbn20)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dt, @michae2, @nameisbhaskar, @RaduBerinde, @rail, and @vidit-bhat)

___
*[`pkg/sql/catalog/tabledesc/validate.go` line 980 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/124372#-NyXTe2q5DEMhuLAiYIQ:-NyXTe2q5DEMhuLAiYIR:b-3ml0ks) ([raw file](https://github.com/cockroachdb/cockroach/blob/ceea4c06f069c7c829f2ae65f6819b73be674b22/pkg/sql/catalog/tabledesc/validate.go#L980)):*
> ```Go
> 	// initialized to validate the storage parameters.
> 	vea.Report(ValidateTTLExpirationExpr(desc))
> 	vea.Report(ValidateTTLExpirationColumn(desc, true /* allowDescPK */))
> ```

nit: this bool param can be removed


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
*: remove all version checks against 23.2

Now that `MinSupported` is `V23_2`, we no longer need to check if 23.2
is active. This change removes all these checks.

Epic: none
Release note: None

RaduBerinde reviewed (commented):
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124372#-:-NyX_-EYESgbjVGigG6A:bz5mair)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @celiala, @dt, @michae2, @nameisbhaskar, @rail, and @vidit-bhat)

___
*[`pkg/sql/catalog/tabledesc/validate.go` line 980 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/124372#-NyXTe2q5DEMhuLAiYIQ:-NyX_1fD0K8JSM41abEQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ceea4c06f069c7c829f2ae65f6819b73be674b22/pkg/sql/catalog/tabledesc/validate.go#L980)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: this bool param can be removed
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
